### PR TITLE
feat: add slsa verification and publish

### LINKED ISSUE #938: Look into adding signed releases into Pepr
### Is your feature request related to a problem? Please describe.

Look into signing Pepr releases

### Describe the solution you'd like

- **Given** a release has been cut
- **When** the key stakeholders look at the release
- **Then** they can tell that it is signed

### Describe alternatives you've considered
(optional) A clear and concise description of any alternative solutions or features you've considered.

### PR DESCRIPTION
**Pull request by cmwylie19**
## Description

Adds slsa to Pepr release process. Now, in `npmjs` this includes only `dist` and `package.json`, LICENSE, and `README.md`, which is what is necessary to run the CLI. I think this is okay. Lets chat further about this.

https://www.npmjs.com/package/peppr?activeTab=code
```bash
┌─[cmwylie19@Cases-MacBook-Pro] - [~/not-pepr/pepr] - [2024-07-29 11:35:08]
└─[0] <git:(938 1a9636b) > npx peppr -V             
0.37.9-rc
```

## Related Issue

Fixes #938 

Relates to #

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request) followed

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -56,4 +56,44 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      - name: Use Node.js latest
+        uses: actions/setup-node@1e60f620b9541d16bece96c5465dc8ee9832be0b # v4.0.3
+        with:
+          node-version: latest
+          cache: "npm"
+      - run: npm ci
+
```
Comment (cmwylie19) [inline .github/workflows/release.yml:69]:
```suggestion
npm run build insert here
```

**Review by naveensrinivasan — DISMISSED** [review]
@cmwylie19  This is amazing! Thank you for taking and running with this.

**Comment by naveensrinivasan** [comment]
@daveworth 👀

File: package.json [hunk 2]
```diff
@@ -9,10 +9,12 @@
   "engines": {
     "node": ">=18.0.0"
   },
+  "files": ["/dist"],
```
Comment (cmwylie19) [inline package.json]:
```suggestion
  "files": ["/dist", "/src"],
```

**Review by schaeferka — APPROVED** [review]
Good as reviewed during Cruising

**Review by btlghrants — APPROVED** [review]

**Pull request merged**